pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -838,19 +843,26 @@ func (gce *GCECloud) GetAutoLabelsForPD(name string, zone string) (map[string]st
 		// We could assume the disks exists; we have all the information we need
 		// However it is more consistent to ensure the disk exists,
 		// and in future we may gather addition information (e.g. disk type, IOPS etc)
-		zoneSet, err := volumeutil.LabelZonesToSet(zone)
-		if err != nil {
-			glog.Warningf("Failed to parse zone field: %q. Will use raw field.", zone)
-		}
-
-		if len(zoneSet) > 1 {
-			// Regional PD
-			disk, err = gce.getRegionalDiskByName(name)
+		if utilfeature.DefaultFeatureGate.Enabled(features.GCERegionalPersistentDisk) {
```

Comment:
Should we be gating this? At this point the disk has already been created. Doing this will just not report the correct info back to the scheduler. But I can buy the argument, that disabling the feature means disabling *all* of it.

Reply:
Without this check, a user can manually provision a PV with with zones label `zone1__zone2` and still have it point to a regional PD. I don't think we should allow this if the flag gate is unset

Reply:
Should it fall back to non-regional behavior in that case, or be completely unusable?

Reply:
The `else` clause here is the non-regional behavior before any regional logic was added